pennylane/estimator/ops/qubit/parametric_ops_single_qubit.py

```diff
@@ -0,0 +1,760 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Resource operators for parametric single qubit operations."""
+
+import numpy as np
+
+from pennylane.estimator.resource_operator import (
+    CompressedResourceOp,
+    GateCount,
+    ResourceOperator,
+    resource_rep,
+)
+from pennylane.exceptions import ResourcesUndefinedError
+
+from ..identity import GlobalPhase
+from .non_parametric_ops import T
+
+# pylint: disable=arguments-differ, signature-differs
+
+
+def _rotation_resources(precision=10e-3):
+    r"""Estimates the number of T gates needed to implement a Pauli rotation to a given precision.
+
+    The expected T-count is taken from the "Simulation Results" section of `Efficient
+    Synthesis of Universal Repeat-Until-Success Circuits <https://arxiv.org/abs/1404.5320>`_.
+    The cost is given as:
+
+        .. math:: T_{count} = \lceil(1.149 * log_{2}(\frac{1}{\epsilon}) + 9.2)\rceil,
+
+    where :math:`\epsilon` is the provided ``precision``.
+
+    Args:
+        precision (float): The acceptable error threshold for the approximation.
+
+    Returns:
+        list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+        where each object represents a specific quantum gate and the number of times it appears
+        in the decomposition.
+    """
+    num_gates = round(1.149 * np.log2(1 / precision) + 9.2)
+    t = resource_rep(T)
+    return [GateCount(t, num_gates)]
+
+
+class PhaseShift(ResourceOperator):
+    r"""Resource class for the PhaseShift gate.
+
+    Args:
+        precision (float, optional): The error threshold for the Clifford + T decomposition
+            of this operation. The default value is ``None`` which corresponds to using the
+            ``precision`` stated in the ``ResourceConfig``.
+        wires (Any or Wires, optional): The wires the operation acts on.
+
+    Resources:
+        The phase shift gate is equivalent to a Z-rotation up to some global phase,
+        as defined from the following identity:
+
+        .. math:: R_\phi(\phi) = e^{i\phi/2}R_z(\phi) = \begin{bmatrix}
+                    1 & 0 \\
+                    0 & e^{i\phi}
+                \end{bmatrix}.
+
+    .. seealso:: The corresponding PennyLane operation :class:`~.pennylane.PhaseShift`.
+
+    **Example**
+
+    The resources for this operation are computed as:
+
+    >>> qml.estimator.PhaseShift.resource_decomp()
+    [(1 x RZ), (1 x GlobalPhase)]
+    """
+
+    num_wires = 1
+    resource_keys = {"precision"}
+
+    def __init__(self, precision=None, wires=None) -> None:
```

Comment:
Import from `pennylane.wires.WiresLike`
```suggestion
    def __init__(self, precision: float | None = None, wires: WiresLike = None) -> None:
```

Reply:
same suggestion for the other __init__ :)